File: mobile/src/main/java/ro/antiprotv/sugar/db/AlertDao.java
package ro.antiprotv.sugar.db;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import java.util.List;

/**
 * The ROOM alert DAO
 */
@Dao
public abstract class AlertDao {

    @Insert
    public abstract long insert(Alert alert);

    /**
     * Retrieves all active alerts
     *
     * @return
     */
    @Query("select * from alert where level != 0 order by level asc")
    public abstract LiveData<List<Alert>> getAllActiveAlerts();

    @Query("select * from alert where level == 0")
    public abstract List<Alert> getAllDisabledAlertsSync();

    @Query("select * from alert where itemName = :itemName")
    public abstract List<Alert> getAlertsByItem(String itemName);


    @Query("update alert set level = 0")
    public abstract void removeAll();

    @Update
    public abstract void updateAlerts(Alert... alerts);

    @Query("select distinct store from alert where store like :prefix order by store")
    public abstract List<String> getStoresByPrefix(String prefix);

    @Query("select distinct store from alert where itemName = :item group by store order by count(store) desc")
    public abstract List<String> selectStoresByItem(String item);
}

File: mobile/src/main/java/ro/antiprotv/sugar/repository/db/AlertDao.kt
package ro.antiprotv.sugar.repository.db

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy.REPLACE
import androidx.room.Query
import androidx.room.Update

/**
 * The ROOM alert DAO
 */
@Dao
interface AlertDao {
    @Insert(onConflict = REPLACE)
    fun insert(alert: Alert): Long

    @Query("select * from alert")
    fun allAlerts(): LiveData<List<Alert>>

    @Query("update alert set type = 0 where type != 0")
    fun removeAll()

    @Update
    fun updateAlerts(vararg alerts: Alert)
}